### PR TITLE
Replace showTreeItemPicker with pickAppResource

### PR DESCRIPTION
**Pull request by alexweininger**
Depends on https://github.com/microsoft/vscode-azuretools/pull/1146 and https://github.com/microsoft/vscode-azureresourcegroups/pull/221

### CODE REVIEW

File: src/postgres/commands/configurePostgresFirewall.ts [hunk 1]
```diff
@@ -16,7 +17,9 @@ import { PostgresServerTreeItem } from "../tree/PostgresServerTreeItem";
 
 export async function configurePostgresFirewall(context: IActionContext, treeItem?: PostgresServerTreeItem): Promise<void> {
     if (!treeItem) {
-        treeItem = <PostgresServerTreeItem>await ext.rgApi.appResourceTree.showTreeItemPicker(PostgresServerTreeItem.contextValue, context);
+        treeItem = await ext.rgApi.pickAppResource<PostgresServerTreeItem>(context, {
+            filter: [postgresSingleFilter, postgresFlexibleFilter]
```
Comment (bwateratmsft) [inline src/postgres/commands/configurePostgresFirewall.ts:21]: Just double checking again, `PostgresServerTreeItem.contextValue` is intentionally dropped?

Comment (bwateratmsft) [reply src/postgres/commands/configurePostgresFirewall.ts:21]: Also this same pattern appears in a couple of the other commands.

**Review by bwateratmsft — APPROVED** [review]

File: src/postgres/commands/configurePostgresFirewall.ts (hunk 1 above)
Comment (nturinski) [reply src/postgres/commands/configurePostgresFirewall.ts:21]: Yeah, the way this works is the filter will give you the `AppResource` that lives on `SubscriptionTreeItem`.  The `expectedContextValue` is only used for searching for children on that `AppResource` so if it's omitted, then it's assumed that the picker ends at the `AppResource`

Comment (alexweininger) [reply src/postgres/commands/configurePostgresFirewall.ts:21]: Yes, the context value is only used for selecting children of top level app resources tree items (AppResource).

**Pull request merged**